src/engine/interop/specification.md

```diff
@@ -0,0 +1,151 @@
+# Engine API. Interop edition
+
+This document specifies a subset of the Engine API methods that are required to be implemented for the Merge interop.
+
+*Note*: Sync API design is yet a draft and considered optional for the interop.
+
+## Underlying protocol
+
+This specification is based on [Ethereum JSON-RPC API](https://eth.wiki/json-rpc/API) and inherits the following properties of this standard:
+* Supported communication protocols (HTTP and WebSocket)
+* Message format and encoding notation
+* [Error codes improvement proposal](https://eth.wiki/json-rpc/json-rpc-error-codes-improvement-proposal)
+
+Client software **MUST** expose Engine API at a port independent from JSON-RPC API. The default port for the Engine API is 8550 for HTTP and 8551 for WebSocket.
+
+## Error codes
+
+The list of error codes introduced by this specification can be found below.
+
+| Code | Possible Return message | Description |
+| - | - | - |
+| 4 | Unknown block | Should be used when a call refers to the unknown block |
+| 5 | Unavailable payload | Should be used when the `payloadId` parameter of `engine_getPayload` call refers to the payload building process that is unavailable |
+
+## Structures
+
+Fields having `DATA` and `QUANTITY` types **MUST** be encoded according to the [HEX value encoding](https://eth.wiki/json-rpc/API#hex-value-encoding) section of Ethereum JSON-RPC API.
+
+*Note:* Byte order of encoded value having `QUANTITY` type is big-endian.
+
+### ExecutionPayload
+
+This structure maps on the [`ExecutionPayload`](https://github.com/ethereum/consensus-specs/blob/dev/specs/merge/beacon-chain.md#ExecutionPayload) structure of the beacon chain spec. The fields are encoded as follows:
+- `parentHash`: `DATA`, 32 Bytes
+- `coinbase`:  `DATA`, 20 Bytes
```

Comment:
```suggestion
- `feeRecipient`:  `DATA`, 20 Bytes
```
Thoughts?

Reply:
I am not opposed to this renaming. If we accept this then the respective change must be done to the beacon chain spec. The `feeRecipient` name stems for the geth codebase which names this field as `Coinbase`, it also stems from JSON-RPC which names the same field as `miner` (in the PoS the latter name doesn't make any sense and we might want to update the JSON-RPC API).

cc @MicahZoltu @djrtwo

Reply:
I'm mixed. Seems easier to not go in and adjust it on all clients but... don't have a dog in this race

Reply:
Actually, we state "The `coinbase` field value **MAY** deviate from what is specified by the `feeRecipient` parameter." in another place. So we would need to be clearer here maybe "`payload.feeRecipient` field value.... from what is specified by the `feeRecipient` call parameter"

Reply:
I have always hated Coinbase as it is totally non-descriptive.  The only thing we know for sure about this address is that it is the address that the consensus client is *suggesting* should receive fees.  Because of this, I favor `suggestedFeeRecipient` to make it very clear to the reader that this is just a suggestion.

Reply:
Sorry, I thought this was for the API.  I agree with `feeRecipient` here, and I think the API method should have `suggestedFeeRecipient`.  That will address @djrtwo's concern about name collisions, and also provide the most descriptive variable names possible (important for specifications IMO).

Reply:
I'd suggest we continue this discussion in a separate issue/PR or discord

Reply:
Yeah, let's take it there. I think the terms are fine for this initial interop release.